apis/project_api/project_api/utils.py

```diff
@@ -194,9 +194,20 @@ def load_scenario_metadata(
     data_dir = list(station_data_dir.glob(f"disagg_{ims[0].file_format()}*"))[0]
     metadata_df = pd.read_csv(list(data_dir.glob("*_metadata.csv"))[0], index_col=0)
 
+    ensemble_disagg_result = gc.disagg.EnsembleDisaggResult.load(data_dir)
+    merged_metadata_df = ensemble_disagg_result.total_contributions_df.merge(
+        metadata_df, how="left", left_index=True, right_index=True
+    )
+    # Drop unnecessary columns and rows
+    merged_metadata_df.drop(labels=["contribution", "epsilon"], axis=1, inplace=True)
+    merged_metadata_df.drop(labels=["distributed_seismicity"], axis=0, inplace=True)
```

Comment:
generally you are better doing `merged_metadata_df = ... ` without the inplace as it is faster. The inplace is really only for when you can't make a copy in the memory available.

In also saying that, you'd be better off doing the drop's as "one line" - I think.

Reply:
drop as `one line` :exploding_head: 
The beauty of Python...